### PR TITLE
Fixes #201 - deprecated traj ctrl

### LINKED ISSUE #201: Deprecated joint spline traj controller for the arm in gazebo
From: http://wiki.ros.org/robot_mechanism_controllers/JointSplineTrajectoryController

Should use http://wiki.ros.org/robot_mechanism_controllers/JointTrajectoryActionController instead.

### PR DESCRIPTION
**Pull request by ugocupcic**
Can't run joint position control + trajectory controllers at the same time anymore. I decided to use trajectory controllers (on the arm) by default when running the arm and hand together as it is more standard. Let me know what you think.

### CODE REVIEW

File: sr_description/arm/config/arm_controller.yaml [hunk 1]
```diff
@@ -34,23 +34,8 @@ sa_er_position_controller:
     d: 5.0
     i_clamp: 5.0
 
-r_arm_cartesian_pose_controller:
```
Comment (ugocupcic) [inline sr_description/arm/config/arm_controller.yaml:37]: doesn't exist in ros-control. Recommended way to do this seems to use moveit compute cartesian path from a program (which uses the trajectory controller to replay the movement).

**Comment by guihomework** [comment]
Just a reminder : The (UPMC) Shadow Arm can only be controlled in position in the real world, and that is the reason a joint_spline_trajectory controller is still in sr_mechanism_controllers. Please do not remove this one, even if you point to the standard one (sending torques) in simulation.

**Comment by ugocupcic** [comment]
@guihomework thanks for the comment! The spline_trajectory controller will be kept in sr_mechanism_controllers. We're just changing this for simulation as you point out.

File: sr_hand/launch/gazebo/gazebo_arm_and_hand.launch [hunk 2]
```diff
@@ -40,15 +40,14 @@
   </group>
   <node name="spawn_hand" pkg="gazebo_ros" type="spawn_model" args="-urdf -param robot_description -z 0.0 -model shadow_model -J ShoulderJSwing 0.78 -J ElbowJSwing 2.0" respawn="false" output="screen" />
 
-  <!-- Controllers for the arm and hand -->
+  <!-- Controllers for the hand -->
   <include file="$(find sr_hand)/launch/gazebo/loaders/hand_controllers.launch"/>
-  <include file="$(find sr_hand)/launch/gazebo/loaders/arm_controllers.launch"/>
 
-  <!-- launch more exotic controllers (trajectory, cartesian, etc...) -->
+  <!-- launch trajectory controllers for the hand -->
```
Comment (toliver) [inline sr_hand/launch/gazebo/gazebo_arm_and_hand.launch]: Should be for the arm?

File: sr_description/arm/config/arm_controller.yaml [hunk 3]
```diff
@@ -34,23 +34,8 @@ sa_er_position_controller:
...
-    i_clamp: 1.0
-  fb_rot:
-    p: 0.5
-    i: 0.1
-    d: 0.0
-    i_clamp: 0.2
-
 r_arm_joint_trajectory_controller:
-  type: robot_mechanism_controllers/JointSplineTrajectoryController
+  type: position_controllers/JointTrajectoryController
```
Comment (toliver) [inline sr_description/arm/config/arm_controller.yaml]: Hi, I was  running it like this, but the arm was not moving.

I've changed this to `type: effort_controllers/JointTrajectoryController` and then they move, but the wrist joints are not.

I've tried to not launch the position controllers for the wrist, but even then, the wrists are not moving (just shaking a bit).

File: sr_description/arm/config/arm_controller.yaml [hunk 4]
```diff
@@ -34,34 +34,15 @@ sa_er_position_controller:
...
     - ElbowJRotate
-    - WRJ2
-    - WRJ1
   gains:
     ShoulderJRotate: {p: 140.0, d: 30.0}
     ShoulderJSwing: {p: 140.0, d: 30.0}
     ElbowJSwing: {p: 140.0, d: 30.0}
     ElbowJRotate: {p: 140.0, d: 30.0}
-    WRJ2: {p: 140.0, d: 30.0}
-    WRJ1: {p: 140.0, d: 30.0}
```
Comment (toliver) [inline sr_description/arm/config/arm_controller.yaml:67]: OK. Tested without the wrist in the controller. Now it works. Happy to merge.

**Pull request merged**